blog/README.md

```diff
@@ -0,0 +1,232 @@
+  CC [M]  /build/simple-kmod/simple-procfs-kmod.o
+  Building modules, stage 2.
+  MODPOST 2 modules
+  CC      /build/simple-kmod/simple-kmod.mod.o
+  LD [M]  /build/simple-kmod/simple-kmod.ko
+  CC      /build/simple-kmod/simple-procfs-kmod.mod.o
+  LD [M]  /build/simple-kmod/simple-procfs-kmod.ko
+make[1]: Leaving directory '/usr/src/kernels/4.18.0-372.53.1.el8_6.x86_64'
+gcc -o spkut ./simple-procfs-kmod-userspace-tool.c
+install -v -m 755 spkut /bin/
+'spkut' -> '/bin/spkut'
+install -v -m 755 -d /lib/modules/4.18.0-372.53.1.el8_6.x86_64/
+install -v -m 644 simple-kmod.ko        /lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-kmod.ko
+'simple-kmod.ko' -> '/lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-kmod.ko'
+install -v -m 644 simple-procfs-kmod.ko /lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-procfs-kmod.ko
+'simple-procfs-kmod.ko' -> '/lib/modules/4.18.0-372.53.1.el8_6.x86_64/simple-procfs-kmod.ko'
+depmod -F /lib/modules/4.18.0-372.53.1.el8_6.x86_64/System.map 4.18.0-372.53.1.el8_6.x86_64
+--> 26dfedc51be
+[2/2] STEP 1/5: FROM quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:127670aaa6073e97bd6d919e1b57bd16b60435410b096aeefce7cd64f6c83d24
+[2/2] STEP 2/5: ARG KERNEL_VERSION
+--> 2879805a870
+[2/2] STEP 3/5: COPY --from=builder /etc/driver-toolkit-release.json /etc/
+--> 8dea5bb2bd8
+[2/2] STEP 4/5: COPY --from=builder /lib/modules/4.18.0-372.53.1.el8_6.x86_64/*.ko /usr/lib/modules/4.18.0-372.53.1.el8_6.x86_64/
+--> 6249995c281
+[2/2] STEP 5/5: RUN depmod -a 4.18.0-372.53.1.el8_6.x86_64 && echo simple_kmod > /etc/modules-load.d/simple_kmod.conf
+[2/2] COMMIT quay.io/ybettan/coreos-layering:simple-kmod
+--> 612c8c8a54f
+Successfully tagged quay.io/ybettan/coreos-layering:simple-kmod
+612c8c8a54f0800cf8dcb4c20d5d3d384dc47463711d6d3ad1bd93d52e982b1e
+
+```
+
+Now, let's push the image to some image registry in order for the image to be accesible
+by the cluster.
+```
+$ podman push quay.io/ybettan/coreos-layering:simple-kmod
+```
+
+Make sure the image is public or it have the correct pull-secret to pull the image.
+The pull secret in the cluster is
+```
+$ oc get secret/pull-secret -n openshift-config.
+```
+
+### Loading the kernel modules into the nodes
+
+First, we will need to create a `MachineConfig` file
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-ybettan-external-image
+spec:
+  osImageURL: quay.io/ybettan/coreos-layering:simple-kmod
+```
+
+Make sure to change the `osImageURL` section in the MachineConfig yaml to the
+image you just built, then apply it.
+```
+$ oc apply -f machineconfig.yaml
+```
+
+The [machine-config-operator](https://github.com/openshift/machine-config-operator)
+will start updating workers nodes one by one with the new image specified in
+`osImageURL` which contain our kernel module.
+
+This may take some time and you can track the progress using
```

Comment:
maybe to specify that the nodes will be rebooted?

Reply:
That's the thing, when I [recorded](https://asciinema.org/a/522655) it few months ago, there was a node reboot but when I wrote this PR I didn't see any reboot.

I don't know if this is a layering enhancement (I know there were plans do avoid reboot when possible) or just an OCP node status that was changed but I didn't see any `Rebooting` state on the nodes.

A quick look into machine-config-operator's logs didn't lead to anything obvious about not rebooting but I will need to dig further.

@cgwalters Has there be any changes regarding MCO not rebooting nodes after applying a MC that uses the container image fitting the image that was on the nodes before applying the MC?

Reply:
Yes, we can support some live updates (at a technical level, `rpm-ostree apply-live` is supported today) but it gets complex and also needs the MCO to learn about this.

Reply:
Yes, we can support some live updates (at a technical level, `rpm-ostree apply-live` is supported today) but it gets complex and also needs the MCO to learn about this.

Reply:
Added a comment about potential nodes reboot.